### PR TITLE
Feature/enable eslint-plugin-mdx

### PR DESCRIPTION
**Pull request by chenxsan**
We have some `mdx` files now, and it's likely that we will migrate all `.md` to `.mdx`, hence enabling `eslint-plugin-mdx` should help catching problems.

Thanks to @JounQin, all issues listed below are gone now.

<del>
There're some issues at the moment:

1. Those code blocks in `.mdx` files won't get linted as `eslint-plugin-mdx` doesn't support it.
2. `vscode-mdx` extension doesn't work as expected in my test.
</del>

### CODE REVIEW

**Comment by montogeek** [comment]
>  and it's likely that we will migrate all .md to .mdx

WHy?

**Comment by chenxsan** [comment]
> > and it's likely that we will migrate all .md to .mdx
> 
> Why?

Because the [`Badge`](https://github.com/webpack/webpack.js.org/tree/master/src/components/Badge) component might be used in other markdown files. And we're maintaining two different configuration for `md` and `mdx` https://github.com/webpack/webpack.js.org/blob/master/webpack.common.js#L57, it's not necessary if `mdx` works for both cases.

**Comment by montogeek** [comment]
> Because the Badge component might be used in other markdown files. And we're maintaining two different configuration for md and mdx https://github.com/webpack/webpack.js.org/blob/master/webpack.common.js#L57, it's not necessary if mdx works for both cases.

Agree, but we shouldn't blindly switch everything when MDX features are only used in a few files of hundreds.
It will also impact the performance of the build.

**Comment by chenxsan** [comment]
> > Because the Badge component might be used in other markdown files. And we're maintaining two different configuration for md and mdx https://github.com/webpack/webpack.js.org/blob/master/webpack.common.js#L57, it's not necessary if mdx works for both cases.
> 
> Agree, but we shouldn't blindly switch everything when MDX features are only used in a few files of hundreds.
> It will also impact the performance of the build.

Don't worry, I won't migrate all to `mdx` at once without any specific reason.

**Comment by JounQin** [comment]
Hi, what do you mean by `vscode-mdx extension doesn't work as expected in my test.`?

Maybe related to https://github.com/mdx-js/vscode-mdx/pull/163?

File: .eslintrc.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ module.exports = {
   plugins: ['markdown', 'cypress', 'react-hooks'],
   globals: {
     __DEV__: true,
+    Badge: true, // global MDX component
```
Comment (JounQin) [inline .eslintrc.js:15]: I think it should be added at `md/mdx` section to affect these files only.

File: .eslintrc.js [hunk 2]
```diff
@@ -26,15 +27,23 @@ module.exports = {
     react: {
       version: 'detect',
     },
+    'mdx/code-blocks': true,
```
Comment (JounQin) [inline .eslintrc.js:30]: Same as above IMO.

**Comment by chenxsan** [comment]
> Hi, what do you mean by `vscode-mdx extension doesn't work as expected in my test.`?
> 
> Maybe related to [mdx-js/vscode-mdx#163](https://github.com/mdx-js/vscode-mdx/pull/163)?

It won't highlight those errors:

image

here's a version of md:

image

File: .eslintrc.js [hunk 3]
```diff
@@ -26,15 +27,23 @@ module.exports = {
...
     {
-      files: ['**/*.{md,mdx}'],
+      files: ['**/*.mdx'],
+      extends: ['plugin:mdx/recommended'],
+      rules: {
+        semi: ['off'],
+      },
+    },
+    {
+      files: ['**/*.md'],
```
Comment (JounQin) [inline .eslintrc.js:42]: It can be linted by `eslint-plugin-mdx` as-is now, with `remark-lint` support.

**Comment by JounQin** [comment]
> It won't highlight those errors:

Yes, try to add `eslint.validate` as https://github.com/mdx-js/vscode-mdx/pull/163/files

File: .eslintrc.js (hunk 1 above)
Comment (chenxsan) [reply .eslintrc.js]: You're right, I've updated it.

**Comment by chenxsan** [comment]
> > It won't highlight those errors:
> 
> Yes, try to add `eslint.validate` as https://github.com/mdx-js/vscode-mdx/pull/163/files

Cool, did fix the problem.

File: .eslintrc.js (hunk 3 above)
Comment (chenxsan) [reply .eslintrc.js:46]: This repository use `markdownlint` at the moment so I'll leave it as it is.

**Comment by chenxsan** [comment]
@JounQin I just notice a warning on my local computer:

```bash
/Users/sam/Documents/githubRepos/webpack.js.org/src/components/NotificationBar/Notification.mdx
  1:94   warning  `"` can be escaped with `&quot;`, `&ldquo;`, `&#34;`, `&rdquo;`  mdx/no-unescaped-entities
  1:95   warning  `>` can be escaped with `&gt;`                                   mdx/no-unescaped-entities
  1:111  warning  `>` can be escaped with `&gt;`                                   mdx/no-unescaped-entities
```
Here's the content of `Notification.mdx`:

```mdx
Webpack 5 has been officially released. Read our <a href="/blog/2020-10-10-webpack-5-release/">announcement</a>. Not ready yet? Read <a href="https://v4.webpack.js.org/">webpack 4 documentation here</a>.
```
Do you have any idea? I think it's valid syntax to use jsx inside markdown?

**Comment by JounQin** [comment]
@chenxsan Please report it to `eslint-mdx`, I can reproduce it.

**Review by JounQin — APPROVED** [review]

**Pull request merged**